src/Maestro/SubscriptionActorService/LocalGit.cs

```diff
@@ -58,51 +58,89 @@ public async Task<string> GetPathToLocalGitAsync()
+                // There are multiple checks whether Git Works, so if it's already there, we'll just rely on 
+                // falling through into CheckGitInstallation()
+                string gitRoot = _tempFiles.GetFilePath("git-portable");
+                string targetPath = Path.Combine(gitRoot, Path.GetFileNameWithoutExtension(remoteFileName));
+                _gitExecutable = Path.Combine(targetPath, "bin", "git.exe");
 
-                        if (Directory.Exists(targetPath))
+                try
+                {
+                    // Determine whether another process/ thread ended up getting the lock and downloaded git in the meantime.
+                    if (!File.Exists(_gitExecutable))
+                    {
+                        using (_operations.BeginOperation($"Installing a local copy of git"))
                         {
-                            _logger.LogInformation($"Target directory {targetPath} already exists. Deleting it.");
-                            Directory.Delete(targetPath, true);
-                        }
+                            string gitZipFile = Path.Combine(gitRoot, remoteFileName);
 
-                        Directory.CreateDirectory(targetPath);
+                            _logger.LogInformation($"Downloading git from '{gitLocation}' to '{gitZipFile}'");
 
-                        using (HttpClient client = new HttpClient())
-                        using (FileStream outStream = new FileStream(gitZipFile, FileMode.Create, FileAccess.Write))
-                        using (var inStream = await client.GetStreamAsync(gitLocation))
-                        {
-                            await inStream.CopyToAsync(outStream);
-                        }
+                            if (Directory.Exists(targetPath))
+                            {
+                                _logger.LogInformation($"Target directory {targetPath} already exists. Deleting it.");
+
+                                // https://github.com/dotnet/arcade/issues/7343
+                                // If this continues to fail despite having both process/thread semaphores wrapping it, we should consider
+                                // killing any git.exe whose image path resides under this folder or simply using another folder and wasting disk space.
+                                Directory.Delete(targetPath, true);
+                            }
 
-                        _logger.LogInformation($"Extracting '{gitZipFile}' to '{targetPath}'");
+                            Directory.CreateDirectory(targetPath);
 
-                        ZipFile.ExtractToDirectory(gitZipFile, targetPath, overwriteFiles: true);
+                            using (HttpClient client = new HttpClient())
+                            using (FileStream outStream = new FileStream(gitZipFile, FileMode.Create, FileAccess.Write))
+                            using (var inStream = await client.GetStreamAsync(gitLocation))
```

Comment:
I think Mutex + await is bad news, because you're going to leave this thread.  Anything in the Mutex needs to be non-async, I think.

Reply:
ah you're right :( my file-based approach wouldn't hit this at all

Reply:
I know. I'm sorry I led you astray.  Do you still have your nifty file option?

Reply:
it doesn't really need to be async here, I would prefer a mutex, and this is on the thread pool so its not blocking anything.

Reply:
After consulting with some folks, it seems like using a named semaphore works and doesn't have trouble.  A lot of times when I try to convert something from synch to async or vice versa I find an API that's not in both places, do you strongly object to just converting to named semaphores?

Reply:
Updated to use named Semaphore after chatting with some SMEs.